dali/kernels/imgproc/convolution/laplacian_gpu.cuh

```diff
@@ -0,0 +1,318 @@
+    }
+
+    KernelContext sub_ctx = ctx;
+    sub_ctx.scratchpad = &sub_scratch;
+
+    // Clear the scratchpad for sub-kernels to reuse memory
+    sobel_dz_.Run(sub_ctx, out, in, windows[0], {scale[0]});
+    sub_scratch.Clear();
+    sobel_dy_.Run(sub_ctx, out, in, windows[1], {scale[1], 1.f});
+    sub_scratch.Clear();
+    sobel_dx_.Run(sub_ctx, out, in, windows[2], {scale[2], 1.f});
+  }
+
+  scratch_sizes_t sub_scratch_sizes_;
+  DzKernel sobel_dz_;
+  DyKernel sobel_dy_;
+  DxKernel sobel_dx_;
+};
+
+
+/**
+ * @brief Laplacian kernel.
+ * Provides separate specialization for 1 dimensional data. For other cases
+ * there are two specializations depending on whether ``Intermediate`` is the same as ``Out``:
+ * if not, the extra intermediate buffer is used to accumulate partial derivatives and then
+ * the result is cast and moved to the output TensorListView.
+ *
+ * For `axes` dimensional input data, there will be computed and summed `axes` partial derivatives,
+ * one in each direction. The `i-th` partial derivative is computed by performing
+ * a `axes`-dimensional separable convolution, where the `i-th` window is responsible for actual
+ * derivative approximation and the remaining windows provide smoothing.
+ * In total, there are ``axes * axes`` windows involved in computing laplacian.
+ *
+ * Dummy type is used with enable_if_t to disambiguate template specializations.
+ */
+template <typename Intermediate, typename Out, typename In, typename W, int axes, bool has_channels,
+          bool is_sequence, typename Dummy = void>
+struct LaplacianGpu;
+
+template <typename Out, typename In, typename W, int axes, bool has_channels, bool is_sequence>
+struct LaplacianGpu<Out, Out, In, W, axes, has_channels, is_sequence, std::enable_if_t<axes != 1>>
+    : LaplacianGpuBase<Out, In, W, axes, has_channels, is_sequence> {};
+
+template <typename Intermediate, typename Out, typename In, typename W, bool has_channels,
+          bool is_sequence>
+struct LaplacianGpu<Intermediate, Out, In, W, 1, has_channels, is_sequence> {
+  static constexpr int axes = 1;
+  using DxKernel = SeparableConvolutionGpu<Out, In, W, axes, has_channels, is_sequence>;
+  static constexpr int ndim = DxKernel::ndim;
+
+  KernelRequirements Setup(
+      KernelContext& ctx, const TensorListShape<ndim>& in_shape,
+      const std::array<std::array<TensorListShape<1>, axes>, axes>& window_sizes,
+      bool has_smoothing) {
+    (void)has_smoothing;
+    return sobel_dx_.Setup(ctx, in_shape, window_sizes[0]);
+  }
+
+  void Run(
+      KernelContext& ctx, const TensorListView<StorageGPU, Out, ndim> out,
+      const TensorListView<StorageGPU, const In, ndim>& in,
+      const std::array<std::array<TensorListView<StorageCPU, const W, 1>, axes>, axes>& windows,
+      std::array<span<const float>, axes> scale) {
+    return sobel_dx_.Run(ctx, out, in, windows[0], {}, {scale[0]});
+  }
+
+  DxKernel sobel_dx_;
```

Comment:
```suggestion
  DxKernel dx_kernel_;
```
While I can pass `PartialDeriv` (or `Laplacian`, for that matter) without built-in facilities for calculating the windows, calling this "Sobel" is a bit too much - there's nothing that makes it Sobel, nor does the implementation need to run sobel filter specifically - it can be any high-pass filter (or derivative).